### DISCUSSION TITLE
Is there a way to set filed level permissions in model settings

### DISCUSSION THREAD

**Discussion by sanojsilva** [strapi/strapi#7563] [Legacy Topics]
I want to hide certain fields from the api based on the user role. I've also checked the documentation but I couldn't find any detail about this. But however in strapi roadmap it shows as already launched.
![image](https://user-images.githubusercontent.com/31297777/90967935-c621fc00-e503-11ea-94a2-4b7be2e34029.png)

**Comment by adrian-ub** [comment]
Hello, regarding what you show is done in the Strapi administration panel, there you can add the permissions per field.

**Reply by sanojsilva** [reply]
Hey, Thanks for the info. Is there a way to do this for the api somehow?

**Comment by derrickmehaffy** [comment]
Hi @sanojsilva for the moment there isn't a simple way to do this but we do have a guide to customize the response, it just requires some custom controller code within your project: https://strapi.io/documentation/v3.x/guides/custom-data-response.html#introduction
